### PR TITLE
Don't validate memory values

### PR DESCRIPTION
**Pull request by Alon-Ti**
Remove validation of values in CairoPie Memory.

### CODE REVIEW

**Review by juanbono — CHANGES_REQUESTED** [review]
Can you add some tests to this?

**Comment by Alon-Ti** [comment]
@juanbono, I want to add a pathological CairoPie end to end test that will cover this and the previous issue against python VM output (see slack), I can do that once #1781 is merged.

**Comment by Alon-Ti** [comment]
@juanbono I amended an existing test to cover this, will add a test CairoPie with all the edge cases once I'm sure all our test CairoPies run.

File: cairo_programs/relocate_segments.cairo [hunk 1]
```diff
@@ -12,14 +12,14 @@ func main() {
...
-    assert array[1] = 51;
+    assert array[1] = cast(array, felt) + 100;
 
     // Realocate temporary_array into the array segment
     relocate_segment(src_ptr=temporary_array, dest_ptr=array);
 
     // Assert that the realocated temporary_array gets their values from the array segment
     assert temporary_array[0] = 50;
-    assert temporary_array[1] = 51;
+    assert temporary_array[1] = cast(array, felt) + 100;
```
Comment (fmoletta) [inline cairo_programs/relocate_segments.cairo:22]: Could we have this as a separate test instead?

**Review by Oppen — APPROVED** [review]

**Review by juanbono — APPROVED** [review]

**Review by fmoletta — APPROVED** [review]
